Allow mounting routeless engines with a bound engine name

mike183 opened the pull request:
This PR builds upon the work in #14967, introducing the ability to pass bound properties to the `{{mount}}` syntax, enabling the dynamic mounting of "routeless" engines.

This also resolves ember-engines/ember-engines#117

cc/ @rwjblue

mike183 committed:
Allow mounting routeless engines with a bound engine name

Assert when an invalid engine name is passed to {{mount}}

Fix {{mount}} tests

rwjblue commented:
LGTM

@dgeb / @chancancode / @trentmwillis - Any objections?

trentmwillis commented on packages/ember-glimmer/lib/syntax/mount.js:
```diff
@@ -72,23 +67,27 @@ class DynamicEngineReference {
-    return this._lastDef;
+      return this._lastDef;
+    } else {
+      return null;
```
IMO this should log if the argument isn't a `string/undefined/null`.

rwjblue replied:
ya, lets have it assert

trentmwillis reviewed (changes requested):
Overall, I'm good with it. For bad input though, I'd like to see feedback given to the user.

mike183 commented:
@trentmwillis Thanks for the review, I've added an assertion to check that the provided engine name is valid and also added a test to go along with it.

cc/ @rwjblue

mike183 commented:
🎉
